runtime/bcverify/vrfyhelp.c

```diff
@@ -907,14 +920,32 @@ isFieldAccessCompatible(J9BytecodeVerificationData *verifyData, J9ROMFieldRef *f
 		J9ROMFieldShape *field = findFieldFromCurrentRomClass(romClass, fieldRef);
 
 #if defined(J9VM_OPT_VALHALLA_STRICT_FIELDS)
-		/* A field declared by the current class with ACC_FINAL and ACC_STRICT flags
-		 * can't be set unless the initialization state is early larval.
-		 */
-		if ((NULL != field)
-			&& J9ROMFIELD_IS_STRICT_FINAL(romClass, field->modifiers)
-			&& (FALSE == liveStack->uninitializedThis)
+		if (J9_CLASSFILE_OR_ROMCLASS_SUPPORTS_STRICT_FIELDS(verifyData->romClass)
+			&& (NULL != field) && J9ROMFIELD_IS_STRICT(field->modifiers)
 		) {
-			return (IDATA)FALSE;
+			/* A strict final field cannot be set in the following cases:
+			* - the initialization state is not early larval
+			* - another instance initialization method has been called
+			*/
+			if (J9_ARE_ALL_BITS_SET(field->modifiers, J9AccFinal)) {
+				if ((FALSE == liveStack->uninitializedThis)
+					|| (TRUE == anotherInstanceInitCalled)
```

Comment:
I think this is a kind of anti-pattern: a boolean value should only be used directly, with unary `!` or compared with `FALSE`, never with `TRUE` (which is only one of a large number of non-false values). Also, less-expensive tests should done first (unless they're expected to be rare).
```c
				if (notherInstanceInitCalled || liveStack->uninitializedThis) {
```
Same comment for line 938